vsts/vsts.yaml

```diff
@@ -1,5 +1,45 @@
 name: $(BuildID)_$(BuildDefinitionName)_$(SourceBranchName)_$(Date:yyyyMMdd)$(Rev:.r)
 
+# https://docs.microsoft.com/azure/devops/pipelines/process/runtime-parameters
+parameters:
+- name: maxParallelJobs
+  displayName: Maximum jobs to run in parallel. Change maxParallel to 1 make builds run in serial rather than in parallel.
+  type: number
+  # Reasonable target to avoid throttling and contention with several PRs running.
+  default: 4
+- name: jobTimeoutInMinutes
+  displayName: Timeout for each job
+  type: number
+  default: 75
+- name: testTargets
+  displayName: The .NET test targets to build and run. 'all' and 'min-matrix' (net6.0, netcore3.1) are aggregates of the remaining values.
+  type: string
+  values:
+    - all
+    - min-matrix
+    - net6.0
+    - net5.0
+    - netcore3.1
+    - netcore2.1
+    - net4.7.2
+    - net4.5.1
+  default: all
+
+variables:
+# Variables for the test matrix based on the 'testTargets' that turn on/off various test targets specifically or based on the 2 aggregates.
+  allTestTargets: ${{ or(contains(parameters['testTargets'], 'all'), ne(variables['Build.Reason'], 'PullRequest') }}
+  minMatrix: ${{ or(or(eq(parameters['testTargets'], 'min-matrix'), eq(variables['allTestTargets'], 'True')), eq(variables['Build.Reason'], 'PullRequest')) }}
+
+# Minimum matrix targets
+  testNet60: ${{ or(eq(variables['minMatrix'], 'True'), contains(parameters['testTargets'], 'net6.0')) }}
+  testNetcore31: ${{ or(eq(variables['minMatrix'], 'True'), contains(parameters['testTargets'], 'netcore3.1')) }}
+
+# Remaining targets
+  testNet50: ${{ or(eq(variables['allTestTargets'], 'True'), contains(parameters['testTargets'], 'net5.0')) }}
+  testNetcore21: ${{ or(eq(variables['allTestTargets'], 'True'), contains(parameters['testTargets'], 'netcore2.1')) }}
+  testNet472: ${{ or(eq(variables['allTestTargets'], 'True'), contains(parameters['testTargets'], 'net4.7.2')) }}
+  testNet451: ${{ or(eq(variables['allTestTargets'], 'True'), contains(parameters['testTargets'], 'net4.5.1')) }}
```

Comment:
How does the yaml know to run allTestTargets or the minMatrix?

Reply:
Intended:

1. Someone specified it when running a build manually.
2. If a PR, it should choose min matrix.
3. If not a PR, it should choose full matrix.

I'm tweaking the logic a bit as I test.